apps/meteor/app/livechat/server/api/lib/livechat.ts

```diff
@@ -62,6 +69,21 @@ export function findGuest(token: string): Promise<ILivechatVisitor | null> {
 	});
 }
 
+export function findGuestBySource(token: string, sourceType: OmnichannelSourceType): Promise<ILivechatVisitor | null> {
```

Comment:
This doesn't return a `ILivechatVisitor`, it returns some fields from it, pls adapt the type

Reply:
All methods we use for sending messages in `livechat/server/v1/messages.ts` (where `findGuest` and `findGuestBySource` are used) require a full `LivechatVisitor`. We probably don't need it (since it's currently working with the incorrect type returned by `findGuest`) , but there'd be A LOT to update if we decide to fix the type returned here
I thought about holding this for another PR, wdyt? That'd almost be a TS conversion on its own :yes-gazzo:

Reply:
Then i'd say this should return a full ILivechatVisitor :(

@MarcosSpessatto wdyt?

Reply:
Couldn't this be dangerous? idk which source of info we hide with the projection :thinking: I see we don't return some stuff such as `ip`, `host`, `livechatData`, `lastAgent`

Reply:
I agree with what @matheusbsilva137 said, let's follow what we have already, if this is a problem, we need to fix it in a subsequent PR.